exonum-java-binding/cryptocurrency-demo/src/test/java/com/exonum/binding/cryptocurrency/transactions/CreateWalletTxTest.java

```diff
@@ -69,60 +67,51 @@ void constructorRejectsNegativeBalance() {
+    assertThat(t.getMessage()).isEqualTo("The initial balance (-1) must not be negative.");
   }
 
   @Test
   @RequiresNativeLibrary
-  void executeCreateWalletTx() throws Exception {
-    CreateWalletTx tx = new CreateWalletTx(DEFAULT_INITIAL_BALANCE);
-
-    try (Database db = MemoryDb.newInstance();
-        Cleaner cleaner = new Cleaner()) {
-      Fork view = db.createFork(cleaner);
-
-      // Execute the transaction
-      TransactionContext context = newContextBuilder(view)
-          .authorPk(OWNER_KEY)
-          .build();
-      tx.execute(context);
-
-      // Check that entries have been added.
-      CryptocurrencySchema schema = new CryptocurrencySchema(view);
-      MapIndex<PublicKey, Wallet> wallets = schema.wallets();
+  void executeCreateWalletTx() {
+    try (TestKit testKit = TestKit.forService(CryptocurrencyServiceModule.class)) {
+      KeyPair emulatedNodeKeyPair = testKit.getEmulatedNode().getServiceKeyPair();
+      TransactionMessage transactionMessage =
+          createCreateWalletTransaction(DEFAULT_INITIAL_BALANCE, emulatedNodeKeyPair);
+      testKit.createBlockWithTransactions(transactionMessage);
+
+      testKit.withSnapshot((view) -> {
+        // Check that entries have been added
+        CryptocurrencySchema schema = new CryptocurrencySchema(view);
+        MapIndex<PublicKey, Wallet> wallets = schema.wallets();
 
-      assertTrue(wallets.containsKey(OWNER_KEY));
-      assertThat(wallets.get(OWNER_KEY).getBalance(), equalTo(DEFAULT_INITIAL_BALANCE));
+        PublicKey emulatedNodePublicKey = emulatedNodeKeyPair.getPublicKey();
+        assertThat(wallets.containsKey(emulatedNodePublicKey)).isTrue();
+        assertThat(wallets.get(emulatedNodePublicKey).getBalance())
+            .isEqualTo(DEFAULT_INITIAL_BALANCE);
+        return null;
+      });
     }
   }
 
   @Test
   @RequiresNativeLibrary
-  void executeAlreadyExistingWalletTx() throws CloseFailuresException {
-    try (Database db = MemoryDb.newInstance();
-        Cleaner cleaner = new Cleaner()) {
-      Fork view = db.createFork(cleaner);
-
-      long initialBalance = DEFAULT_INITIAL_BALANCE;
-
-      // Create a wallet manually.
-      CryptocurrencySchema schema = new CryptocurrencySchema(view);
-      {
-        MapIndex<PublicKey, Wallet> wallets = schema.wallets();
-        wallets.put(OWNER_KEY, new Wallet(initialBalance));
-      }
-
-      // Execute the transaction, that has the same owner key.
-      // Use twice the initial balance to detect invalid updates.
-      long newBalance = 2 * initialBalance;
-      CreateWalletTx tx = new CreateWalletTx(newBalance);
-      TransactionContext context = newContextBuilder(view)
-          .authorPk(OWNER_KEY)
-          .build();
-      TransactionExecutionException e = assertThrows(
-          TransactionExecutionException.class, () -> tx.execute(context));
-
-      assertThat(e.getErrorCode(), equalTo(WALLET_ALREADY_EXISTS.errorCode));
+  void executeAlreadyExistingWalletTx() {
```

Comment:
This code lacks some comments and structure to make the scenario easy to understand. E.g.,

```java
      // Create a new wallet
      TransactionMessage transactionMessage =
          createCreateWalletTransaction(DEFAULT_INITIAL_BALANCE, OWNER_KEY_PAIR);
      testKit.createBlockWithTransactions(transactionMessage);

      // Attempt to execute a transaction with the same owner public key.
      // Use different balance so that it is not rejected as a duplicate
      TransactionMessage transactionMessage2 =
          createCreateWalletTransaction(DEFAULT_INITIAL_BALANCE * 2, OWNER_KEY_PAIR);
      testKit.createBlockWithTransactions(transactionMessage2);

      // Check the second tx failed
      testKit.withSnapshot((view) -> {
        Blockchain blockchain = Blockchain.newInstance(view);
        Optional<TransactionResult> txResult = blockchain.getTxResult(transactionMessage2.hash());
        TransactionResult expectedTransactionResult =
            TransactionResult.error(WALLET_ALREADY_EXISTS.errorCode, null);
        assertThat(txResult).hasValue(expectedTransactionResult);
        return null;
      });
```